api_app/script_analyzers/observable_analyzers/emailrep.py

```diff
@@ -0,0 +1,42 @@
+import requests
+
+from api_app.exceptions import AnalyzerRunException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+class EmailRep(classes.ObservableAnalyzer):
+    base_url: str = "https://emailrep.io/{}"
+
+    def set_config(self, additional_config_params):
+        self.api_key_name = additional_config_params.get("api_key_name", "EMAILREP_KEY")
+        self.__api_key = secrets.get_secret(self.api_key_name)
+
+    def run(self):
+        """
+        API key is not mandatory, emailrep supports requests with no key:
+        a valid key let you to do more requests per day.
+        therefore we're not checking if a key has been configured.
+        """
+
+        headers = {
+            "User-Agent": "IntelOwl v2",
+            "Key": self.__api_key,
```

Comment:
in case the `self.__api_key` is `None` (case when it's not defined by user), does passing a null API key to the URL does not give error ?

Reply:
No, It did not give any errors with `None`.

Reply:
as far as i tested, an empty key doesn't give any error